### PR TITLE
[Change] ] Vanilla Templates, added additional DLC conditionals 

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Utilizing enabledDLC switches for DLC vehicles and weapons
Tweaking inconsistencies between templates

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

Loaded each template, checked RPT for non-obvious errors.

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
Check DLC boxes, Load template, observe changes.
Provide opinion on changes to the vehicle lineups.

********************************************************
Notes:
Changes:
1. [x] DLC conditionals for vehicles
2. [x] Matched vehicle templates (ie Nato tropical only used HMG mrap while others used both)
3. [x] Moved DLC vehicles who had possible non-dlc standins to relevant DLC toggle*
4. [x] CSAT DLC weaponry.
5. [x] NATO DLC weaponry.
6. [x] AAF DLC weaponry.
7. [x] LDF DLC weaponry.

*Please review and give feedback on this point in particular.

### CODE REVIEW

**Comment by Lazejun** [comment]
This is a fun one.

So this WILL cause the problem that people that missuse the DLC Parameters for more Uniforms etc. causes more DLC items to be used.

You can kill the DLC Map Specific Templates in the process, just a good question what the fallback will be for those that used them in their saves.

--------> Sort of People problem, do what you think is right, i will point everone complaining towards your direction.

https://community.bistudio.com/wiki/Arma_3:_DLC_Restrictions

**Comment by ante185** [comment]
> 
> So this WILL cause the problem that people that missuse the DLC Parameters for more Uniforms etc. causes more DLC items to be used.
> 
To my knowledge the only DLC options that currently add uniforms for the player is Apex and contact, which are not usable unless you have the DLC anyways, not quite sure what it is that you mean

> You can kill the DLC Map Specific Templates in the process, just a good question what the fallback will be for those that used them in their saves.
> 
Think you could elaborate a little bit on this?

**Comment by ante185** [comment]
Changes complete, review needed.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf [hunk 1]
```diff
@@ -29,7 +29,7 @@ private _cargoTrucks = ["O_T_Truck_02_transport_F","O_T_Truck_02_F","O_T_Truck_0
 ["vehiclesLightAPCs", []] call _fnc_saveToTemplate;
 ["vehiclesAPCs", ["O_T_APC_Tracked_02_cannon_ghex_F","O_T_APC_Wheeled_02_rcws_v2_ghex_F"]] call _fnc_saveToTemplate;
 ["vehiclesIFVs", []] call _fnc_saveToTemplate;
-["vehiclesTanks", ["O_T_MBT_02_cannon_ghex_F"]] call _fnc_saveToTemplate;
+private _Tanks = ["O_T_MBT_02_cannon_ghex_F"] call _fnc_saveToTemplate;
```
Comment (jaj22) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf:32]: bug

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf [hunk 2]
```diff
@@ -386,6 +399,43 @@ _pilotLoadoutData set ["uniforms", ["U_O_PilotCoveralls"]];
...
+	["srifle_DMR_04_F", "", "acc_pointer_IR", "optic_DMS", [], [], "bipod_02_F_hex"]
+	];
+	_sfSniper append [
+	["srifle_DMR_05_hex_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_KHS_blk", [], [], "bipod_02_F_hex"],
+	["srifle_DMR_05_blk_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_LRPS", [], [], "bipod_02_F_hex"]
+	];
+	
+	_sfLoadoutData set ["machineGuns", _sfMG];
+	_sfLoadoutData set ["marksmanRifles", _sfMarks];
+	_sfLoadoutData set ["sniperRifles", _sfSniper];
```
Comment (jaj22) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf:422]: You can do these in-place:
```
	private _sfMG      = _sfLoadoutData get "machineGuns";
	_sfMG append [
	["MMG_01_tan_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_Arco", [], [], "bipod_02_F_hex"],
	["MMG_01_hex_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_MRCO", [], [], "bipod_02_F_hex"]
	];
	_sfLoadoutData set ["machineGuns", _sfMG];
```
Does the same thing as:
```
	(_sfLoadoutData get "machineGuns") append [
	["MMG_01_tan_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_Arco", [], [], "bipod_02_F_hex"],
	["MMG_01_hex_F", "muzzle_snds_93mmg_tan", "acc_pointer_IR", "optic_MRCO", [], [], "bipod_02_F_hex"]
	];
```

Comment (ante185) [reply A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf:422]: I didn't know that the get operator got by reference, neat

Comment (ante185) [reply A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Apex.sqf]: Applied

**Comment by Lazejun** [comment]
Let me guess this PR is basically dead as all of its changes are in 2914?

**Comment by ante185** [comment]
> Let me guess this PR is basically dead as all of its changes are in 2914?

yeah no, but kind of yes?
Having thought about it i believe that i will have to redo and divide these things a bit.

**Pull request closed without merging**